mmv1/third_party/terraform/tests/resource_datapipeline_pipeline_test.go

```diff
@@ -0,0 +1,349 @@
+          zone                            = "us-centra1-a"
+          service_account_email           = google_service_account.service_account.email
+          network                         = "default"
+          temp_location                   = "gs://my-bucket/tmp_dir"
+          machine_type                    = "E2"
+          additional_experiments          = []
+          additional_user_labels = {
+            "context" : "test"
+          }
+          worker_region    = "us-central1"
+          worker_zone      = "us-central1-a"
+
+          enable_streaming_engine = "false"
+        }
+        container_spec_gcs_path = "gs://my-bucket/template"
+        update                  = false
+        transform_name_mappings = {"name": "wrench"}
+      }
+      location = "us-central1"
+    }
+  }
+  schedule_info {
+    schedule  = "0 * * * *"
+    time_zone = "UTC"
+  }
+  pipeline_sources = {
+    "name": "wrench"
+  }
+}
+`
+}
+
+func testAccDataPipelinePipeline_basicLaunchTemplateUpdate() string {
+	return `
+resource "google_service_account" "service_account" {
+  account_id   = "service-account-id"
+  display_name = "Service Account"
+}
+
+resource "google_data_pipeline_pipeline" "primary" {
+  name         = "tf-test-pipeline"
+  display_name = "update-pipeline"
+  type         = "PIPELINE_TYPE_BATCH"
+  state        = "STATE_ACTIVE"
+
+  workload {
+    dataflow_launch_template_request {
+      project_id = "my-project"
+      gcs_path = "gs://my-bucket/path"
+      launch_parameters {
+        job_name = "my-job"
+        parameters = {
+          "name": "wrench"
+        }
+        environment {
+          num_workers                     = 5
+          max_workers                     = 5
+          zone                            = "us-centra1-a"
+          service_account_email           = google_service_account.service_account.email
+          network                         = "default"
+          temp_location                   = "gs://my-bucket/tmp_dir"
+          bypass_temp_dir_validation      = false
+          machine_type                    = "E2"
+          additional_experiments          = ["test"]
+          additional_user_labels = {
+            "context" : "test"
+          }
+          worker_region    = "us-central1"
+          worker_zone      = "us-central1-a"
+
+          enable_streaming_engine = "false"
+        }
+        update                  = false
+        transform_name_mapping = {"name": "wrench"}
+      }
+      location = "us-central1"
+    }
+  }
+  schedule_info {
+    schedule  = "0 * * * *"
+    time_zone = "UTC"
+  }
+  pipeline_sources = {
+    "name": "wrench"
+  }
+}
+`
+}
+func testAccDataPipelinePipeline_basicLaunchTemplate() string {
+	return `
+resource "google_service_account" "service_account" {
+  account_id   = "service-account-id"
+  display_name = "Service Account"
+}
+
+resource "google_data_pipeline_pipeline" "primary" {
+  name         = "tf-test-pipeline"
```

Comment:
This name needs to have a randomized postfix
```suggestion
  name         = "tf-test-pipeline%{random_suffix}"
```

Reply:
see https://github.com/hashicorp/terraform-provider-google-beta/blob/bf4be67fea5143544711c13ad0f026cc997cbac3/google-beta/resource_beyondcorp_app_connection_test.go#L52